trunk/SUAVE/Plots/Geometry_Plots/plot_vehicle.py

```diff
@@ -39,286 +40,582 @@ def plot_vehicle(vehicle, save_figure = False, plot_control_points = True, save_
-        dim_R    = len(VD.Wake.XA1[0,0,0,:])
-        for p_idx in range(num_prop): 
-            for t_idx in range(nts): 
-                for B_idx in range(num_B):
-                    for loc in range(dim_R): 
-                        X = [VD.Wake.XA1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.XB1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.XB2[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.XA2[p_idx,t_idx,B_idx,loc]]
-                        Y = [VD.Wake.YA1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.YB1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.YB2[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.YA2[p_idx,t_idx,B_idx,loc]]
-                        Z = [VD.Wake.ZA1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.ZB1[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.ZB2[p_idx,t_idx,B_idx,loc],
-                             VD.Wake.ZA2[p_idx,t_idx,B_idx,loc]]                    
-                        verts = [list(zip(X, Y, Z))]
-                        collection = Poly3DCollection(verts)
-                        collection.set_facecolor(face_color)
-                        collection.set_edgecolor(edge_color) 
-                        collection.set_alpha(alpha)
-                        axes.add_collection3d(collection)  
-                        
-    if np.all(VD.FUS_SURF_PTS) != None:   
-        face_color = 'grey'                
-        edge_color = 'black' 
-        alpha      = 1
+        # store points
+        plot_fuselage_geometry(axes,fus_pts, face_color,edge_color,alpha)
         
-        num_fus_segs = len(VD.FUS_SURF_PTS[:,0,0])
-        tesselation  = len(VD.FUS_SURF_PTS[0,:,0]) 
-        for i_seg in range(num_fus_segs-1):
-            for i_tes in range(tesselation-1):
-                X = [VD.FUS_SURF_PTS[i_seg  ,i_tes  ,0],
-                     VD.FUS_SURF_PTS[i_seg  ,i_tes+1,0],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes+1,0],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes  ,0]]
-                Y = [VD.FUS_SURF_PTS[i_seg  ,i_tes  ,1],
-                     VD.FUS_SURF_PTS[i_seg  ,i_tes+1,1],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes+1,1],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes  ,1]]
-                Z = [VD.FUS_SURF_PTS[i_seg  ,i_tes  ,2],
-                     VD.FUS_SURF_PTS[i_seg  ,i_tes+1,2],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes+1,2],
-                     VD.FUS_SURF_PTS[i_seg+1,i_tes  ,2]]                 
-                verts = [list(zip(X, Y, Z))]
-                collection = Poly3DCollection(verts)
-                collection.set_facecolor(face_color)
-                collection.set_edgecolor(edge_color) 
-                collection.set_alpha(alpha)
-                axes.add_collection3d(collection)  
+    return 
+
+  
+def plot_fuselage_geometry(axes,fus_pts, face_color,edge_color,alpha):  
```

Comment:
think of a better name, it's a little confusing between this function and the last

Reply:
Name changed to "generate_fuselage_surface"